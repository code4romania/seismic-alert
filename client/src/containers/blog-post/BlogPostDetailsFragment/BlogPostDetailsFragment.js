import React from 'react';
import { useParams } from 'react-router-dom';
import { Col, Row, Tag, Typography } from 'antd';
import { Trans } from '@lingui/macro';
import useWindowSize from '../../../hooks/useWindowSize';

import config from '../../../config';

const { Title, Paragraph } = Typography;
const { POSTS_URL } = config;

const BlogPostDetailsFragment = () => {
  const { slug } = useParams();
  const windowSize = useWindowSize();
  const [state, setState] = React.useState({
    postDetails: null,
    requestError: false,
  });
  React.useEffect(() => {
    fetch(`${POSTS_URL}/${slug}`)
      .then((res) => (res.status === 200 ? res.json() : null))
      .then((postDetails) => {
        setState((prevState) => ({
          ...prevState,
          postDetails,
          requestError: postDetails === null,
        }));
      })
      .catch(() => {
        setState((prevState) => ({
          ...prevState,
          postDetails: null,
          requestError: true,
        }));
      });
  }, []);

  if (state.postDetails === null) {
    return state.requestError ? (
      <p>
        <Trans>Article not found</Trans>
      </p>
    ) : (
      <p />
    );
  }
  const authorFullName = `${state.postDetails.author_first_name} ${state.postDetails.author_last_name}`;

  return (
    <Row
      type="flex"
      justify="center"
      align="top"
      style={{ marginTop: '2rem', marginBottom: '2rem' }}
    >
      <Col span={16}>
        <Title level={3} style={{ textTransform: 'uppercase', textAlign: 'left' }}>
          {state.postDetails.title}
        </Title>
        <Paragraph style={{ textAlign: 'left' }}>
          <Trans>
            Published by {authorFullName} â€¢{' '}
            {new Date(state.postDetails.created).toLocaleDateString()}
          </Trans>
        </Paragraph>
        <Paragraph style={{ textAlign: 'left' }}>
          {state.postDetails.tags.map((tag) => (
            <Tag key={tag}>#{tag}</Tag>
          ))}
        </Paragraph>
        <Paragraph style={{ textAlign: 'justify' }}>
          {/* eslint-disable-next-line react/no-danger */}
          <div dangerouslySetInnerHTML={{ __html: state.postDetails.text }} />
        </Paragraph>
        <Paragraph>
          <img
            src={state.postDetails.image}
            width="100%"
            style={{ maxHeight: windowSize.height }}
          />
        </Paragraph>
      </Col>
    </Row>
  );
};

export default BlogPostDetailsFragment;
