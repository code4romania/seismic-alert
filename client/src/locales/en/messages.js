/*eslint-disable*/module.exports={messages:{"About":"About","Add a building":"Add a building","Address":"Address","All seismic risk classes":"All seismic risk classes","An independent, non-partisan, non-political, non-govermental organization.":"An independent, non-partisan, non-political, non-govermental organization.","Article not found":"Article not found","Building Info":"Building Info","Certified expert name":"Certified expert name","Check here if a building is on the seismic risk list":"Check here if a building is on the seismic risk list","Code of conduct":"Code of conduct","Consolidated buildings":"Consolidated buildings","Construction Year":"Construction Year","Contact us":"Contact us","DONATE":"DONATE","Download":"Download","Evaluated buildings":"Evaluated buildings","Examination year":"Examination year","Guide":"Guide","Height regime":"Height regime","Homeowners associations guide":"Homeowners associations guide","Information missing":"Information missing","Legislation":"Legislation","Next page":"Next page","Other":"Other","Page {pageNumber} of {numPages}":["Page ",["pageNumber"]," of ",["numPages"]],"People under risk":"People under risk","Previous page":"Previous page","Privacy policy":"Privacy policy","Published by {authorFullName} • {0}":["Published by ",["authorFullName"]," • ",["0"]],"Risk category":"Risk category","Search Results":"Search Results","Subscribe to the newsletter":"Subscribe to the newsletter","Terms & conditions":"Terms & conditions","Total surface":"Total surface","Useful links":"Useful links","Vulnerable Bucharest":"Vulnerable Bucharest"}};