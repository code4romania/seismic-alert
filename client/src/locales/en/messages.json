{
  "About": {
    "origin": [
      [
        "components/Header.js",
        63
      ]
    ],
    "extractedComments": []
  },
  "Add a building": {
    "origin": [
      [
        "src/components/Header.js",
        41
      ]
    ],
    "obsolete": true
  },
  "Address": {
    "origin": [
      [
        "components/BuildingDetails.js",
        25
      ]
    ],
    "extractedComments": []
  },
  "All seismic risk classes": {
    "origin": [
      [
        "containers/home/MapFragment.js",
        50
      ]
    ],
    "extractedComments": []
  },
  "An independent, non-partisan, non-political, non-govermental organization.": {
    "origin": [
      [
        "components/Footer.js",
        93
      ]
    ],
    "extractedComments": []
  },
  "Article not found": {
    "origin": [
      [
        "containers/blog-post/BlogPostDetailsFragment/BlogPostDetailsFragment.js",
        41
      ]
    ],
    "extractedComments": []
  },
  "Building Info": {
    "origin": [
      [
        "components/BuildingDetails.js",
        19
      ]
    ],
    "extractedComments": []
  },
  "Certified expert name": {
    "origin": [
      [
        "components/BuildingDetails.js",
        61
      ]
    ],
    "extractedComments": []
  },
  "Check here if a building is on the seismic risk list": {
    "origin": [
      [
        "containers/home/SearchFragment.js",
        53
      ]
    ],
    "extractedComments": []
  },
  "Code of conduct": {
    "origin": [
      [
        "components/Footer.js",
        56
      ]
    ],
    "extractedComments": []
  },
  "Consolidated buildings": {
    "origin": [
      [
        "containers/home/StatisticFragment.js",
        49
      ]
    ],
    "extractedComments": []
  },
  "Construction Year": {
    "origin": [
      [
        "components/BuildingDetails.js",
        31
      ]
    ],
    "extractedComments": []
  },
  "Contact us": {
    "origin": [
      [
        "components/Header.js",
        73
      ]
    ],
    "extractedComments": []
  },
  "DONATE": {
    "origin": [
      [
        "components/SocialBar/SocialBar.js",
        87
      ]
    ],
    "extractedComments": []
  },
  "Download": {
    "origin": [
      [
        "containers/guide/index.js",
        36
      ]
    ],
    "extractedComments": []
  },
  "Evaluated buildings": {
    "origin": [
      [
        "containers/home/StatisticFragment.js",
        44
      ]
    ],
    "extractedComments": []
  },
  "Examination year": {
    "origin": [
      [
        "components/BuildingDetails.js",
        55
      ]
    ],
    "extractedComments": []
  },
  "Guide": {
    "origin": [
      [
        "components/Header.js",
        68
      ]
    ],
    "extractedComments": []
  },
  "Height regime": {
    "origin": [
      [
        "components/BuildingDetails.js",
        37
      ]
    ],
    "extractedComments": []
  },
  "Homeowners associations guide": {
    "origin": [
      [
        "containers/guide/index.js",
        22
      ]
    ],
    "extractedComments": []
  },
  "Information missing": {
    "origin": [
      [
        "components/BuildingDetails.js",
        68
      ]
    ],
    "extractedComments": []
  },
  "Legislation": {
    "origin": [
      [
        "src/components/Header.js",
        31
      ]
    ],
    "obsolete": true
  },
  "Next page": {
    "origin": [
      [
        "containers/guide/PdfViewer.js",
        89
      ]
    ],
    "extractedComments": []
  },
  "Other": {
    "origin": [
      [
        "containers/home/MapFragment.js",
        59
      ]
    ],
    "extractedComments": []
  },
  "Page {pageNumber} of {numPages}": {
    "origin": [
      [
        "containers/guide/PdfViewer.js",
        80
      ]
    ],
    "extractedComments": []
  },
  "People under risk": {
    "origin": [
      [
        "containers/home/StatisticFragment.js",
        39
      ]
    ],
    "extractedComments": []
  },
  "Previous page": {
    "origin": [
      [
        "containers/guide/PdfViewer.js",
        86
      ]
    ],
    "extractedComments": []
  },
  "Privacy policy": {
    "origin": [
      [
        "components/Footer.js",
        49
      ]
    ],
    "extractedComments": []
  },
  "Published by {authorFullName} â€¢ {0}": {
    "origin": [
      [
        "containers/blog-post/BlogPostDetailsFragment/BlogPostDetailsFragment.js",
        61
      ]
    ],
    "extractedComments": []
  },
  "Risk category": {
    "origin": [
      [
        "components/BuildingDetails.js",
        49
      ]
    ],
    "extractedComments": []
  },
  "Search Results": {
    "origin": [
      [
        "components/SearchResults.js",
        29
      ]
    ],
    "extractedComments": []
  },
  "Subscribe to the newsletter": {
    "origin": [
      [
        "components/Footer.js",
        77
      ]
    ],
    "extractedComments": []
  },
  "Terms & conditions": {
    "origin": [
      [
        "components/Footer.js",
        42
      ]
    ],
    "extractedComments": []
  },
  "Total surface": {
    "origin": [
      [
        "components/BuildingDetails.js",
        43
      ]
    ],
    "extractedComments": []
  },
  "Useful links": {
    "origin": [
      [
        "components/Footer.js",
        37
      ]
    ],
    "extractedComments": []
  },
  "Vulnerable Bucharest": {
    "origin": [
      [
        "src/components/Header.js",
        34
      ]
    ],
    "obsolete": true
  }
}