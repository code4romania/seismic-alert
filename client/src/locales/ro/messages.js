/*eslint-disable*/module.exports={messages:{"About":"Despre","Add a building":"Adaugă o clădire","Address":"Adresa","All seismic risk classes":"Toate clasele de risc seismic","An independent, non-partisan, non-political, non-govermental organization.":"Organizație neguvernamentală independentă, neafiliată politic și apolitică.","Article not found":"Articolul nu a fost găsit","Building Info":"Informații Clădire","Certified expert name":"Numele expertului evaluator","Check here if a building is on the seismic risk list":"Verifică aici dacă o clădire se află pe lista de risc seismic","Code of conduct":"Codul de conduită","Consolidated buildings":"Clădiri consolidate","Construction Year":"Anul construcției","Contact us":"Contact","DONATE":"DONEAZĂ","Download":"Descarcă","Evaluated buildings":"Clădiri evaluate","Examination year":"Anul expertizei seismice","Guide":"Ghid","Height regime":"Regimul de înălțime","Homeowners associations guide":"Ghid pentru asociații de proprietari","Information missing":"Fără informații","Legislation":"Legislație","Next page":"Pagina următoare","Other":"Altele","Page {pageNumber} of {numPages}":["Pagina ",["pageNumber"]," din ",["numPages"]],"People under risk":"Oameni sub risc","Previous page":"Pagina anterioară","Privacy policy":"Politica de confidențialitate","Published by {authorFullName} • {0}":["Publicat de ",["authorFullName"]," • ",["0"]],"Risk category":"Clasa de risc","Search Results":"Rezultate Căutare","Subscribe to the newsletter":"Abonează-te la newsletter","Terms & conditions":"Termeni şi condiţii","Total surface":"Suprafața desfășurată","Useful links":"Link-uri utile","Vulnerable Bucharest":"Bucureștiul vulnerabil"}};