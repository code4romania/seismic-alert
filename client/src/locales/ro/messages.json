{
  "About": {
    "translation": "Despre",
    "origin": [
      [
        "components/Header.js",
        63
      ]
    ],
    "extractedComments": []
  },
  "Add a building": {
    "translation": "Adaugă o clădire",
    "origin": [
      [
        "src/components/Header.js",
        41
      ]
    ],
    "obsolete": true
  },
  "Address": {
    "translation": "Adresa",
    "origin": [
      [
        "components/BuildingDetails.js",
        25
      ]
    ],
    "extractedComments": []
  },
  "All seismic risk classes": {
    "translation": "Toate clasele de risc seismic",
    "origin": [
      [
        "containers/home/MapFragment.js",
        50
      ]
    ],
    "extractedComments": []
  },
  "An independent, non-partisan, non-political, non-govermental organization.": {
    "translation": "Organizație neguvernamentală independentă, neafiliată politic și apolitică.",
    "origin": [
      [
        "components/Footer.js",
        93
      ]
    ],
    "extractedComments": []
  },
  "Article not found": {
    "translation": "Articolul nu a fost găsit",
    "origin": [
      [
        "containers/blog-post/BlogPostDetailsFragment/BlogPostDetailsFragment.js",
        41
      ]
    ],
    "extractedComments": []
  },
  "Building Info": {
    "translation": "Informații Clădire",
    "origin": [
      [
        "components/BuildingDetails.js",
        19
      ]
    ],
    "extractedComments": []
  },
  "Certified expert name": {
    "translation": "Numele expertului evaluator",
    "origin": [
      [
        "components/BuildingDetails.js",
        61
      ]
    ],
    "extractedComments": []
  },
  "Check here if a building is on the seismic risk list": {
    "translation": "Verifică aici dacă o clădire se află pe lista de risc seismic",
    "origin": [
      [
        "containers/home/SearchFragment.js",
        53
      ]
    ],
    "extractedComments": []
  },
  "Code of conduct": {
    "translation": "Codul de conduită",
    "origin": [
      [
        "components/Footer.js",
        56
      ]
    ],
    "extractedComments": []
  },
  "Consolidated buildings": {
    "translation": "Clădiri consolidate",
    "origin": [
      [
        "containers/home/StatisticFragment.js",
        49
      ]
    ],
    "extractedComments": []
  },
  "Construction Year": {
    "translation": "Anul construcției",
    "origin": [
      [
        "components/BuildingDetails.js",
        31
      ]
    ],
    "extractedComments": []
  },
  "Contact us": {
    "translation": "Contact",
    "origin": [
      [
        "components/Header.js",
        73
      ]
    ],
    "extractedComments": []
  },
  "DONATE": {
    "translation": "DONEAZĂ",
    "origin": [
      [
        "components/SocialBar/SocialBar.js",
        87
      ]
    ],
    "extractedComments": []
  },
  "Download": {
    "translation": "Descarcă",
    "origin": [
      [
        "containers/guide/index.js",
        36
      ]
    ],
    "extractedComments": []
  },
  "Evaluated buildings": {
    "translation": "Clădiri evaluate",
    "origin": [
      [
        "containers/home/StatisticFragment.js",
        44
      ]
    ],
    "extractedComments": []
  },
  "Examination year": {
    "translation": "Anul expertizei seismice",
    "origin": [
      [
        "components/BuildingDetails.js",
        55
      ]
    ],
    "extractedComments": []
  },
  "Guide": {
    "translation": "Ghid",
    "origin": [
      [
        "components/Header.js",
        68
      ]
    ],
    "extractedComments": []
  },
  "Height regime": {
    "translation": "Regimul de înălțime",
    "origin": [
      [
        "components/BuildingDetails.js",
        37
      ]
    ],
    "extractedComments": []
  },
  "Homeowners associations guide": {
    "translation": "Ghid pentru asociații de proprietari",
    "origin": [
      [
        "containers/guide/index.js",
        22
      ]
    ],
    "extractedComments": []
  },
  "Information missing": {
    "translation": "Fără informații",
    "origin": [
      [
        "components/BuildingDetails.js",
        68
      ]
    ],
    "extractedComments": []
  },
  "Legislation": {
    "translation": "Legislație",
    "origin": [
      [
        "src/components/Header.js",
        31
      ]
    ],
    "obsolete": true
  },
  "Next page": {
    "translation": "Pagina următoare",
    "origin": [
      [
        "containers/guide/PdfViewer.js",
        89
      ]
    ],
    "extractedComments": []
  },
  "Other": {
    "translation": "Altele",
    "origin": [
      [
        "containers/home/MapFragment.js",
        59
      ]
    ],
    "extractedComments": []
  },
  "Page {pageNumber} of {numPages}": {
    "translation": "Pagina {pageNumber} din {numPages}",
    "origin": [
      [
        "containers/guide/PdfViewer.js",
        80
      ]
    ],
    "extractedComments": []
  },
  "People under risk": {
    "translation": "Oameni sub risc",
    "origin": [
      [
        "containers/home/StatisticFragment.js",
        39
      ]
    ],
    "extractedComments": []
  },
  "Previous page": {
    "translation": "Pagina anterioară",
    "origin": [
      [
        "containers/guide/PdfViewer.js",
        86
      ]
    ],
    "extractedComments": []
  },
  "Privacy policy": {
    "translation": "Politica de confidențialitate",
    "origin": [
      [
        "components/Footer.js",
        49
      ]
    ],
    "extractedComments": []
  },
  "Published by {authorFullName} • {0}": {
    "translation": "Publicat de {authorFullName} • {0}",
    "origin": [
      [
        "containers/blog-post/BlogPostDetailsFragment/BlogPostDetailsFragment.js",
        61
      ]
    ],
    "extractedComments": []
  },
  "Risk category": {
    "translation": "Clasa de risc",
    "origin": [
      [
        "components/BuildingDetails.js",
        49
      ]
    ],
    "extractedComments": []
  },
  "Search Results": {
    "translation": "Rezultate Căutare",
    "origin": [
      [
        "components/SearchResults.js",
        29
      ]
    ],
    "extractedComments": []
  },
  "Subscribe to the newsletter": {
    "translation": "Abonează-te la newsletter",
    "origin": [
      [
        "components/Footer.js",
        77
      ]
    ],
    "extractedComments": []
  },
  "Terms & conditions": {
    "translation": "Termeni şi condiţii",
    "origin": [
      [
        "components/Footer.js",
        42
      ]
    ],
    "extractedComments": []
  },
  "Total surface": {
    "translation": "Suprafața desfășurată",
    "origin": [
      [
        "components/BuildingDetails.js",
        43
      ]
    ],
    "extractedComments": []
  },
  "Useful links": {
    "translation": "Link-uri utile",
    "origin": [
      [
        "components/Footer.js",
        37
      ]
    ],
    "extractedComments": []
  },
  "Vulnerable Bucharest": {
    "translation": "Bucureștiul vulnerabil",
    "origin": [
      [
        "src/components/Header.js",
        34
      ]
    ],
    "obsolete": true
  }
}