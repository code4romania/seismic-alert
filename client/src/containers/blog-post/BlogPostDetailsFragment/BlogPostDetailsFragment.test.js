import React from 'react';
import { render, cleanup, act } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { i18n } from '@lingui/core';
import { I18nProvider } from '@lingui/react';
import BlogPostDetailsFragment from './BlogPostDetailsFragment';
import { messages as enMessages } from '../../../locales/en/messages';
import { messages as roMessages } from '../../../locales/ro/messages';

i18n.load({
  en: enMessages,
  ro: roMessages,
});

const TestingProvider = ({ children }) => <I18nProvider i18n={i18n}>{children}</I18nProvider>;

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    slug: 'slug',
  }),
}));

jest.mock('../../../hooks/useWindowSize', () => ({
  __esModule: true,
  default: () => ({
    height: 768,
  }),
}));
afterEach(cleanup);

describe('BlogPostDetailsFragment component', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  act(() => {
    i18n.activate('ro');
  });

  it('should render correctly', async () => {
    const blogPost = {
      title: 'Title',
      text: '<p>Text</p>',
      created: new Date(0),
      tags: ['tag1', 'tag2'],
      image: 'image-url',
      author_first_name: 'First',
      author_last_name: 'Last',
    };
    jest.spyOn(global, 'fetch').mockResolvedValue({
      json: jest.fn().mockResolvedValue(blogPost),
      status: 200,
    });
    const container = render(<BlogPostDetailsFragment />, { wrapper: TestingProvider });
    await container.findByText('Title');
    expect(container.baseElement).toMatchSnapshot();
  });

  it('should render correctly if error', async () => {
    jest.spyOn(global, 'fetch').mockResolvedValue({
      json: jest.fn().mockResolvedValue(null),
      status: 404,
    });
    const container = render(<BlogPostDetailsFragment />, { wrapper: TestingProvider });
    await container.findByText('Articolul nu a fost gÄƒsit');
    expect(container.baseElement).toMatchSnapshot();
  });
});
