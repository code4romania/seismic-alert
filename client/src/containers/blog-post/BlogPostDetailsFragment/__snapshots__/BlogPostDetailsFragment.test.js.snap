// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostDetailsFragment component should render correctly 1`] = `
<body>
  <div>
    <div
      class="ant-row-flex ant-row-flex-center ant-row-flex-top"
      style="margin-top: 2rem; margin-bottom: 2rem;"
    >
      <div
        class="ant-col ant-col-16"
      >
        <h3
          class="ant-typography"
          style="text-transform: uppercase; text-align: left;"
        >
          Title
        </h3>
        <div
          class="ant-typography"
          style="text-align: left;"
        >
          Publicat de First Last • 1/1/1970
        </div>
        <div
          class="ant-typography"
          style="text-align: left;"
        >
          <span
            class="ant-tag"
          >
            #
            tag1
          </span>
          <span
            class="ant-tag"
          >
            #
            tag2
          </span>
        </div>
        <div
          class="ant-typography"
          style="text-align: justify;"
        >
          <div>
            <p>
              Text
            </p>
          </div>
        </div>
        <div
          class="ant-typography"
        >
          <img
            src="image-url"
            style="max-height: 768px;"
            width="100%"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`BlogPostDetailsFragment component should render correctly if error 1`] = `
<body>
  <div>
    <p>
      Articolul nu a fost găsit
    </p>
  </div>
</body>
`;
